ext/drreg/drreg.c

```diff
@@ -173,15 +184,57 @@ get_where_app_pc(instr_t *where)
  * SPILLING AND RESTORING
  */
 
+static ptr_uint_t drreg_note_base;
+
+/* Get note value for a drreg instr note. */
+static inline ptr_uint_t
+get_drreg_note_val(uint val)
+{
+    return (ptr_uint_t)(drreg_note_base + val);
+}
+
+/* Returns whether the given ilist has an existing usage of the given slot
+ * on or after `where`. Such a usage, if any, must have been added by a
+ * previous instrumentation pass, and tells us that it is not safe to reuse
+ * the slot in the current pass.
+ */
+static bool
+has_pending_slot_usage_by_prior_pass(per_thread_t *pt, instrlist_t *ilist, instr_t *where,
+                                     uint slot)
+{
+    if (!TEST(DRREG_HANDLE_MULTI_PHASE_SLOT_RESERVATIONS, pt->bb_props))
+        return false;
+    for (instr_t *in = where; in != NULL /*sanity*/ && in != instrlist_last(ilist);
```

Comment:
Please comment on why this loop does not look at the last instr as this is an unusual iteration.

Reply:
(I thought I fixed this before, apparently not). Done.